widget_driver_generator/lib/src/utils/element_utils.dart

```diff
@@ -21,6 +21,14 @@ class ElementUtils {
     return null;
   }
 
+  /// Checks if the current `element` has an annotation of the provided `validAnnotationType`.
+  /// Returns `true`, if this true and `false` accordingly.
+  bool hasValidAnnotation({
+    required Element element,
+    required Type validAnnotationType,
+  }) =>
+      TypeChecker.fromRuntime(validAnnotationType).hasAnnotationOfExact(element);
```

Comment:
There is already a function which checks if an annotation is valid.
could you reuse that function here.. And call it, instead of adding your own logic?
So sort of like this:

```
bool hasValidAnnotation({
    required Element element,
    required Type validAnnotationType,
  }) => getValidAnnotation(element: element, validAnnotationTypes: [validAnnotationType]) != null
```

Reply:
Or the other way around.. Have the `getValidAnnotation` call your method..
But we should probably not have 2 implementations which have logic for how to check the annotation..